### ISSUE TITLE
docker-compose up fails with fresh git cloned dkron

### ISSUE THREAD

**Issue by randallsquared** [victorcoder/dkron#348]
## Version

https://github.com/victorcoder/dkron/tree/b20e67bb452e1ee22770c2734453ed74820a0cce

(master as of April 17)

## Reproduce

```
git clone https://github.com/victorcoder/dkron.git
# edit out zk and etcd from docker-compose.yaml (probably not necessary, but it's what I did)
docker-compose up
# lots of output, then...

dkron_1   | builtin/bins/dkron-processor-files/files_output.go:8:2: cannot find package "github.com/Sirupsen/logrus" in any of:
dkron_1   | 	/usr/local/go/src/github.com/Sirupsen/logrus (from $GOROOT)
dkron_1   | 	/gopath/src/github.com/Sirupsen/logrus (from $GOPATH)
[...]
dkron_1   | dkron/config.go:14:2: cannot find package "github.com/spf13/viper" in any of:
dkron_1   | 	/usr/local/go/src/github.com/spf13/viper (from $GOROOT)
dkron_1   | 	/gopath/src/github.com/spf13/viper (from $GOPATH)
dkron_1   | scripts/entrypoint.sh: line 7: /gopath/src/github.com/victorcoder/dkron/main: No such file or directory
dkron_dkron_1 exited with code 127

```

## Workaround

The root problem appears to be the volume in the docker-compose.yaml overwriting the results of the Dockerfile's `dep ensure`.  Commenting out the volume makes it go.  I think the reason this has worked previously for developers is probably that they had already built locally, meaning that the volume source contained previously ensured deps.

Another workaround is to do `dep ensure` in `entrypoint.sh` ahead of the build, which takes quite a while the first time, but would allow devs and evaluators like me to use the same docker-compose.yaml, since everyone would end up with a local copy of the binary.

**Comment by jaimebrolesi** [comment]
I had another problem here, I don't if its related but I got this error when trying to download and use the docker image:

```
time="2018-04-25T01:21:28Z" level=info msg="No valid config found: Applying default values." error="Config File \"dkron\" Not Found in \"[/etc/dkron /root/.dkron /opt/local/dkron/config]\""

time="2018-04-25T01:21:28Z" level=info msg="agent: Dkron agent starting" node=fcd59d8f5a20

Starting Dkron agent...

time="2018-04-25T01:21:28Z" level=info msg="agent: joining: [] replay: true" node=fcd59d8f5a20

time="2018-04-25T01:21:28Z" level=fatal msg="store: Store backend not reachable" error="client: etcd cluster is unavailable or misconfigured; error #0: dial tcp 127.0.0.1:2379: getsockopt: connection refused\n" node=fcd59d8f5a20
```

**Comment by victorcoder** [comment]
Seems to be solved in v0.10.0
